src/plzinit/plugins.go

```diff
@@ -0,0 +1,285 @@
+		"defaultdbgcflags":   "DefaultDbgCFlags",
+		"defaultoptcppflags": "DefaultOptCppFlags",
+		"defaultdbgcppflags": "DefaultDbgCppFlags",
+		"defaultldflags":     "DefaultLdFlags",
+		"pkgconfigpath":      "PkgConfigPath",
+		"testmain":           "TestMain",
+		"dsymtool":           "DsymTool",
+	}
+
+	return writeFieldsToConfig("cc", file, configMap)
+}
+
+func writeJavaConfigFields(file ast.File) ast.File {
+	configMap := map[string]string{
+		"javactool":          "JavacTool",
+		"javacworker":        "JavacWorker",
+		"junitrunner":        "JunitRunner",
+		"defaulttestpackage": "DefaultTestPackage",
+		"releaselevel":       "ReleaseLevel",
+		"targetlevel":        "TargetLevel",
+		"javacflags":         "JavacFlags",
+		"javactestflags":     "JavacTestFlags",
+		"defaultmavenrepo":   "MavenRepo",
+		"toolchain":          "Toolchain",
+	}
+
+	subsection := "java"
+	section := "Plugin"
+
+	// Check for existing java fields first
+	for _, s := range file.Sections {
+		if s.Key == subsection {
+			for _, field := range s.Fields {
+				if plugVal, ok := configMap[strings.ToLower(field.Name)]; ok {
+					file = ast.InjectField(file, plugVal, field.Value, section, subsection, false)
+				}
+			}
+		}
+	}
+
+	return file
+}
+
+func writeFieldsToConfig(plugin string, file ast.File, configMap map[string]string) ast.File {
+	section := "Plugin"
+	foundSection := false
+	// Check for existing cc fields first and migrate values
+	for _, s := range file.Sections {
+		if s.Key == plugin {
+			foundSection = true
+			for _, field := range s.Fields {
+				if plugVal, ok := configMap[strings.ToLower(field.Name)]; ok {
+					file = ast.InjectField(file, plugVal, field.Value, section, plugin, false)
+				}
+			}
+		}
+	}
+
+	// If we found nothing, add a section with default values commented out
+	if !foundSection {
+		config, err := core.ReadDefaultConfigFiles(nil)
+		if err != nil {
+			panic(err)
+		}
+
+		// Build plugin target so we can pull the default values
+		state := core.NewBuildState(config)
+		buildLabel := core.NewBuildLabel("plugins", plugin)
+		plz.Run([]core.BuildLabel{buildLabel}, nil, state, config, state.TargetArch)
+		subrepo := state.Graph.Subrepo(plugin)
+		if subrepo == nil {
+			log.Fatalf("failed to get subrepo %v", plugin)
+		}
+		if err = subrepo.LoadSubrepoConfig(); err != nil {
+			panic(err)
+		}
+		config = subrepo.State.Config
+		for _, v := range config.PluginConfig {
+			file = ast.InjectField(file, "; "+v.ConfigKey, v.DefaultValue[0], section, plugin, false)
+		}
+	}
+
+	return file
+}
+
+// targetExistsInFile checks to see if the plugin target already exists
+// in plugins/BUILD
+func targetExistsInFile(location, plugin string) bool {
+	if _, err := os.Stat(location); err != nil {
+		log.Debug("Target already exists in %s", location)
+		return false
+	}
+
+	b, err := ioutil.ReadFile(location)
+	if err != nil {
+		panic(err)
+	}
+
+	str := "plugin_repo\\(...name = \"" + plugin + "\""
```

Comment:
Ideally, we would query the graph to check for things like this, was this not possible?

Reply:
I think that would be a better approach and more resilient. If you know what needs to be built then it should be straightforward and fast enough since there are only so many plugins.